plans/sep6.md

```diff
@@ -0,0 +1,255 @@
+}
+
+interface NonInteractiveKycNeeded extends TransferResponse {
+  type: "non_interactive_customer_info_needed";
+  fields: string[]; // This can be restricted to the list of strings in SEP-9
+  // https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0009.md#kyc--aml-fields
+}
+
+interface InteractiveKycNeeded extends TransferResponse {
+  type: "interactive_customer_info_needed";
+  url: string;
+  interactiveDeposit?: boolean;
+}
+
+interface KycStatus extends TransferResponse {
+  type: "customer_info_status";
+  more_info_url?: string;
+  eta?: number;
+}
+
+interface CustomerInfoStatus extends KycStatus {
+  status: "pending" | "denied";
+}
+
+interface KycPromptStatus extends KycStatus {
+  status: "success" | "pending" | "denied";
+}
+
+interface Field {
+  name: string;
+  description: string;
+  optional?: boolean;
+  choices?: [string];
+}
+
+interface Fee {
+  type: "none" | "simple" | "complex";
+}
+
+interface SimpleFee extends Fee {
+  type: "simple";
+  fixed: number;
+  percent: number;
+}
+
+interface Memo {
+  type: "text" | "id" | "hash";
+  value: string;
+}
+```
+
+### Usage
+
+```js
+import { startWithdrawal, startDeposit, RESPONSE_TYPES } from "wallet-sdk";
+
+const withdrawProvider = new WithdrawProvider(transferServerUrl);
+const depositProvider = new DepositProvider(transferServerUrl);
+
+// Retrieve supported assets. Used to display options to the user, including
+// things like simple fees, asset codes, and withdrawal types.
+await withdrawProvider.fetchSupportedAssets();
+await depositProvider.fetchSupportedAssets();
+
+// Returns a single number of how much the user will pay, in units of the asset
+fee = await withdrawProvider.fetchFinalFee({
+  assetCode,
+  amount,
+  type,
+});
+fee = await depositProvider.fetchFinalFee({
+  assetCode,
+  amount,
+  type,
+});
+
+/**
+ * Deposit just needs to display information to the user for them to complete on
+ * their own, but withdraw needs to actually submit a transaction, so the code
+ * paths are a little bit different.
+ */
+const depositResult = await depositProvider.withdraw({
+  asset,
+  destination,
+  etc,
+});
+const withdrawResult = await withdrawProvider.withdraw({
+  asset,
+  destination,
+  etc,
+});
+
+switch (depositResult.type /* or withdrawResult */) {
+  case RESPONSE_TYPES.ok:
+    // if deposit
+    showUser(depositResult);
+
+    // if withdraw, need to sign/submit transaction
+    submitPayment({
+      memo: withdrawResult.memo,
+      destination: withdrawResult.accountId,
+      amount,
+    });
+    break;
+  case RESPONSE_TYPES.interactiveKyc:
+    if (isBrowser) {
+      // To avoid popup blockers, the new window has to be opened directly in
+      // response to a user click event, so we need consumers to provide us a
+      // window instance that they created previously.
+      const popup = window.open("", "name", "dimensions etc");
+
+      const kycResult = await withdrawProvider.kycPrompt(withdrawResult, popup);
```

Comment:
Can you add these functions above to the class definition, so they're documented? Otherwise they could be easy to miss!

I think the function name can be better, both to be unambiguously a verb, and to hit that it needs a window instance. And I feel like I'd think that the popup would be the first argument, so maybe named object prop args are better here.